core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java

```diff
@@ -1741,8 +1744,17 @@ public String getSubclassForDiscriminatorValue(Object value) {
 		return spaces;
 	}
 
-	public TupleContext getTupleContext() {
-		return tupleContext;
+	/**
+	 * Returns the {@link TupleContext}.
+	 *
+	 * @param session the current session, if null the {@link TupleContext#getTransactionContext()} will be null.
+	 * @return the tupleContext for the session
+	 */
+	public TupleContext getTupleContext( SessionImplementor session ) {
```

Comment:
I'm wondering why you opted to require the `SeesionImplementor` in all those code paths (the ones invoking this method) when you're needing the transaction context.

Woudln't it have been possible to invoke that TransactionContextHelper less frequently?